app/src/main/java/com/jerboa/model/PostsViewModel.kt

```diff
@@ -15,38 +15,37 @@ import com.jerboa.api.toApiState
 import com.jerboa.datatypes.BanFromCommunityData
 import com.jerboa.db.entity.AnonAccount
 import com.jerboa.db.repository.AccountRepository
-import com.jerboa.findAndUpdatePost
-import com.jerboa.findAndUpdatePostCreator
-import com.jerboa.findAndUpdatePostCreatorBannedFromCommunity
-import com.jerboa.mergePosts
+import com.jerboa.feed.PaginationController
+import com.jerboa.feed.PostController
 import com.jerboa.toEnumSafe
 import it.vercruysse.lemmyapi.dto.ListingType
 import it.vercruysse.lemmyapi.dto.SortType
 import it.vercruysse.lemmyapi.v0x19.datatypes.CreatePostLike
 import it.vercruysse.lemmyapi.v0x19.datatypes.DeletePost
 import it.vercruysse.lemmyapi.v0x19.datatypes.FeaturePost
 import it.vercruysse.lemmyapi.v0x19.datatypes.GetPosts
-import it.vercruysse.lemmyapi.v0x19.datatypes.GetPostsResponse
 import it.vercruysse.lemmyapi.v0x19.datatypes.LockPost
 import it.vercruysse.lemmyapi.v0x19.datatypes.MarkPostAsRead
-import it.vercruysse.lemmyapi.v0x19.datatypes.PaginationCursor
 import it.vercruysse.lemmyapi.v0x19.datatypes.PersonView
 import it.vercruysse.lemmyapi.v0x19.datatypes.PostView
 import it.vercruysse.lemmyapi.v0x19.datatypes.SavePost
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.launch
 
 open class PostsViewModel(protected val accountRepository: AccountRepository) : ViewModel() {
-    var postsRes: ApiState<GetPostsResponse> by mutableStateOf(ApiState.Empty)
-        private set
-    private var page by mutableLongStateOf(1)
-    protected var pageCursor: PaginationCursor? by mutableStateOf(null)
+    val lazyListState = LazyListState()
+    var postsRes: ApiState<List<PostView>> by mutableStateOf(ApiState.Empty)
         private set
     var sortType by mutableStateOf(SortType.Active)
         private set
     var listingType by mutableStateOf(ListingType.Local)
         private set
 
+    private val pageController = PaginationController()
+    private val postController = PostController()
+
+    private val postsList = postController.feed
```

Comment:
> Doesn't this need to be public to get the benefits of the snapshotstate list

`SnapshotStateList` implements List. Its type does not need to be explicitly `SnapshotStateList`

To benefit from it (This can be verified by observing the the recompositions. Else the post simply wouldnt update since the list did not change. And since we declared it stable. Compose wouldn't see any change and thus not recompose but since its actually a SnapShotStateList it emits a event just for that item and the post does recomposition)

> It seems safer to keep one copy of this list, rather than two.

I'll remove postsList and update PostController to only expose List

I don't want it to be modified other than by the Controller itself.

Reply:
Sounds good.